src/main/kotlin/com/expediagroup/sdk/core/configuration/provider/FileSystemConfigurationProvider.kt

```diff
@@ -35,6 +35,7 @@ internal class FileSystemConfigurationProvider(
 
     override val key: String? by lazy { credentials[Key.CLIENT_KEY] }
     override val secret: String? by lazy { credentials[Key.CLIENT_SECRET] }
-    override val endpoint: String? by lazy { configurations[Key.ENDPOINT] }
+    override val endpoint: String? by lazy { configurations[Key.ENDPOINT]?.let { appendSlashIfRequired(it) } }
     override val authEndpoint: String? by lazy { configurations[Key.AUTH_ENDPOINT] }
+    private fun appendSlashIfRequired(string: String) = if ('/' != string.last()) "$string/" else string
```

Comment:
_nit_: Please add the return type and a new line between the new function and the previous values.